airframe-http-recorder/src/main/scala/wvlet/airframe/http/recorder/HttpRecordStore.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package wvlet.airframe.http.recorder
+
+import java.time.Instant
+import java.util.concurrent.atomic.AtomicInteger
+
+import com.twitter.finagle.http.{Request, Response}
+import wvlet.airframe.jdbc.{DbConfig, SQLiteConnectionPool}
+import wvlet.log.LogSupport
+
+/**
+  * Recorder for HTTP server responses
+  */
+class HttpRecordStore(val recorderConfig: HttpRecorderConfig, dropSession: Boolean = false)
+    extends AutoCloseable
+    with LogSupport {
+  private val connectionPool = new SQLiteConnectionPool(DbConfig.ofSQLite(recorderConfig.sqliteFilePath))
+
+  private def recordTableName = recorderConfig.recordTableName
+
+  // Prepare a database table for recording HttpRecord
+  connectionPool.executeUpdate(HttpRecord.createTableSQL(recordTableName))
+  connectionPool.executeUpdate(
+    s"create index if not exists ${recordTableName}_index on ${recordTableName} (session, requestHash)")
+  // TODO: Detect schema change
+  if (dropSession) {
+    warn(s"Deleting old session records for session:${recorderConfig.sessionName}")
+    connectionPool.executeUpdate(s"delete from ${recordTableName} where session = '${recorderConfig.sessionName}'")
+  }
+
+  private val requestCounter = scala.collection.mutable.Map.empty[Int, AtomicInteger]
+
+  def resetCounter: Unit = {
+    requestCounter.clear()
+  }
+
+  def find(request: Request): Option[HttpRecord] = {
```

Comment:
`findNext` might be better in this case to show this method has side effect.

Reply:
Agreed. I'll change the name